lib/rspec/matchers/dsl.rb

```diff
@@ -80,11 +77,20 @@ def match(&match_block)
         # @yield [Object] actual the actual value (i.e. the value wrapped by `expect`)
         def match_when_negated(&match_block)
           define_user_override(:does_not_match?, match_block) do |actual|
-            @actual = actual
-            super(*actual_arg_for(match_block))
+            begin
+              @actual = actual
+              RSpec::Support.with_failure_notifier(RAISE_METHOD) do
```

Comment:
See https://relishapp.com/rspec-staging/rspec-expectations/docs/aggregating-failures for the feature @JonRowe mentioned.  See cc66c173b148ef666271359c8885ac4555f7d473 for the change that went into that commit.  The specs there show a case where this logic is necessary.  It's good you've added this here but we're going to want a spec for it as well.  Can you add one based on https://github.com/rspec/rspec-expectations/blob/cc66c173b148ef666271359c8885ac4555f7d473/spec/rspec/matchers/dsl_spec.rb#L605-L623 ?